src/main/java/org/killbill/billing/client/KillBillClient.java

```diff
@@ -1220,13 +1220,14 @@ public InvoiceItem createExternalCharge(final InvoiceItem externalCharge, @Nulla
     }
 
     public List<InvoiceItem> createExternalCharges(final Iterable<InvoiceItem> externalCharges, @Nullable final LocalDate requestedDate, final Boolean autoPay, final Boolean autoCommit, final String createdBy, final String reason, final String comment) throws KillBillClientException {
-        return createExternalCharges(externalCharges, requestedDate, autoPay, autoCommit, RequestOptions.builder()
-                                                                                            .withCreatedBy(createdBy)
-                                                                                            .withReason(reason)
-                                                                                            .withComment(comment).build());
+        return createExternalCharges(externalCharges, requestedDate, autoPay, autoCommit, null, null,
+                                     RequestOptions.builder()
+                                                   .withCreatedBy(createdBy)
+                                                   .withReason(reason)
+                                                   .withComment(comment).build());
     }
 
-    public List<InvoiceItem> createExternalCharges(final Iterable<InvoiceItem> externalCharges, final LocalDate requestedDate, final Boolean autoPay, final Boolean autoCommit, final RequestOptions inputOptions) throws KillBillClientException {
+    public List<InvoiceItem> createExternalCharges(final Iterable<InvoiceItem> externalCharges, final LocalDate requestedDate, final Boolean autoPay, final Boolean autoCommit, final String paymentExternalKey, final String transactionExternalKey, final RequestOptions inputOptions) throws KillBillClientException {
```

Comment:
Could you also keep a version with `RequestOptions` without having to pass the keys (since it's the new API moving forward)?